### ISSUE TITLE
extend CBC status_map_mip

### ISSUE THREAD

**Issue by BastiaanvG** [cvxgrp/cvxpy#800]
First of all, I really like working with cvxpy, it makes working with different solvers so much easier!

The issue I have is related to cbc and to this issue: https://github.com/cvxgrp/cvxpy/issues/585 of setting MIP stopping criteria

If the solver quits on maximumSeconds, or any other stopping criteria CyCBC accepts, the cbcModel returns a status not specified in the STATUS_MAP_MIP dictionary resulting in a key not found error. 

```
    prob.solve(solver= 'CBC', verbose=True, allowablePercentageGap =10)
  File "C:\Python36\lib\site-packages\cvxpy\problems\problem.py", line 289, in solve
    return solve_func(self, *args, **kwargs)
  File "C:\Python36\lib\site-packages\cvxpy\problems\problem.py", line 571, in _solve
    self, data, warm_start, verbose, kwargs)
  File "C:\Python36\lib\site-packages\cvxpy\reductions\solvers\solving_chain.py", line 194, in solve_via_data
    solver_opts, problem._solver_cache)
  File "C:\Python36\lib\site-packages\cvxpy\reductions\solvers\conic_solvers\cbc_conif.py", line 177, in solve_via_data
    solution["status"] = self.STATUS_MAP_MIP[status]
KeyError: 'stopped on gap'
```

stopping on maximumSeconds returns: *'stopped on time '*
stopping on allowableGap returns: *'stopped on gap'*
stopping on allowablePercentageGap returns:  *'stopped on gap'*
stopping on maximumNodes returns: *'stopped in nodes'*

There is one problem, however,  stopping on time does not guarantee a feasible solution but the status message does not hold this info it simply returns stopped on time. 

I could extend the STATUS_MAP_MIP dictionary in cbc_conif.py and make a pull request if you think this would be a useful feature.

**Comment by SteveDiamond** [comment]
There should never be a key not found error. You'll need to map all the statuses into the existing status codes. You can always use solver error if it's not clear what status is appropriate. But to address your question, yes, a contribution would be welcome!

**Comment by SteveDiamond** [comment]
Resolved in #802 by @BastiaanvG
